examples/robot_localization_failure/custom_learners/image_based_lightning_learner.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import annotations
+
+import logging
+import os
+import platform
+from collections.abc import Callable
+from io import BytesIO
+from typing import Any
+
+import lightning
+import polars as pl
+import torch
+from feature_images import FeatureImagesData, FeatureImagesPredictionData
+from lightning.pytorch.callbacks.early_stopping import EarlyStopping
+from torch.utils.data import DataLoader, Dataset
+from typing_extensions import override
+
+from flowcean.core import Model, SupervisedLearner
+
+DatasetFactory = Callable[[pl.DataFrame], Dataset]
+
+logger = logging.getLogger(__name__)
+
+
+class ImageBasedLightningLearner(SupervisedLearner):
```

Comment:
I think this can be generalized to the default lightning learner in flowcean, and removed here.